zrml/prediction-markets/src/lib.rs

```diff
@@ -564,16 +603,27 @@ mod pallet {
         /// * `weights`: The relative denormalized weight of each outcome asset. The sum of the
         ///     weights must be less or equal to _half_ of the `MaxTotalWeight` constant of the
         ///     swaps pallet.
+        ///
+        /// # Weight
+        ///
+        /// Complexity:
+        /// - buy_complete_set: `O(n)`,
+        /// where `n` is the number of outcome assets for the categorical market.
+        /// - deploy_swap_pool_for_market_open_pool: `O(n)`,
+        /// where `n` is the number of outcome assets for the categorical market.
+        /// - deploy_swap_pool_for_market_future_pool: `O(n + m)`,
+        /// where `n` is the number of outcome assets for the categorical market,
+        /// and `m` is the number of market ids,
+        /// which open at the same time as the specified market.
         #[pallet::weight(
             T::WeightInfo::buy_complete_set(T::MaxCategories::get().into())
-            .saturating_add(T::WeightInfo::deploy_swap_pool_for_market(
-                T::MaxCategories::get().into(),
-            ))
-            // Overly generous estimation, since we have no access to Swaps WeightInfo
-            // (it is loosely coupled to this pallet using a trait). Contains weight for
-            // create_pool() and swap_exact_amount_in()
-            .saturating_add(5_000_000_000.saturating_mul(T::MaxCategories::get().into()))
-            .saturating_add(T::DbWeight::get().reads(2 as Weight))
+            .saturating_add(
+                T::WeightInfo::deploy_swap_pool_for_market_open_pool(weights.len() as u32)
+            .max(
+                T::WeightInfo::deploy_swap_pool_for_market_future_pool(
+                    weights.len() as u32, CacheSize::get()
+                ))
+            )
```

Comment:
Same here. Closing a parens at the end of the line is is definitely not rustfmt compliant. Again, I feel silly for cracking down on this, but if rustfmt _could_ verify correct formatting here, the CI would reject this code.

Reply:
😂 I didn't make the effort to check it with rustfmt. I just formatted it by hand how it would look good. I am sure, if the CI can check that in the future, the `rustfmt` could format it too! Do you require me to change it here?

Reply:
Let's leave it the way it is, for now. ;)